airflow/providers/tabular/provider.yaml

```
@@ -31,6 +31,7 @@ versions:
 
 dependencies:
   - apache-airflow>=2.4.0
+  - pyiceberg==0.4.0
```

Wondering whether there is a specific reason we need to pin it to 0.4.0?

---

PyIceberg is still quite a fresh project, so I would suggest using 0.4.0 as a lower bound.
```suggestion
  - pyiceberg>=0.4.0
```